k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -1381,6 +1381,10 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
+        if(getName().equalsIgnoreCase("INBOX")){
+            String inbox = new String("INBOX") ;
+            return inbox.hashCode();
```

Comment:
Just do

"INBOX".hashCode();

Nice approach though 👍 

Should be good to push once you've tidied that.